differentiate-json/main.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+  Copyright (C) 2022,  icgc-argo
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU Affero General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+  Authors:
+    Edmund Su
+"""
+
+import os
+import sys
+import argparse
+import json
+import numpy
+
+
+def main():
+    """
+    Python implementation of tool: differentiate-json
+
+    This is auto-generated Python code, please update as needed!
+    """
+
+    parser = argparse.ArgumentParser(description='differentiate JSON metadata payload for SONG upload')
+    parser.add_argument('-b', '--auto_generated', dest="auto_generated", help="auto generated json", required=True)
+    parser.add_argument('-a', '--user_provided',dest="user_provided", help="user generated json", required=True)
+
+    results = parser.parse_args()
+
+    with open(results.auto_generated) as json_file:
+            ag_dict = json.load(json_file)
+    with open(results.user_provided) as json_file:
+            up_dict = json.load(json_file)
+            
+    warnings=[]
+    errors=[]
+    exceptions=[
+            "sample_barcode",
+            "insert_size",
+            "read_group_id_in_bam",
+            "legacyAnalysisId"
+            ]
+    check_values(up_dict,ag_dict,warnings,errors,[],exceptions)
+
+    if len(warnings)>0:
+        with open('WARNINGS.log', 'w') as f:
+            for warning in warnings:
```

Comment:
When will we get `WARNINGS.log`?

Reply:
I haven't coded for any `WARNING` scenarios yet.